### PR TITLE
Add FileUpload Serializer

### PR DESCRIPTION
**Pull request by nick-benoit14**
This adds support for serializing and desalinating file upload questions. The file upload question is extremely basic, so at this point at least, it can just rely on a base serializer and deserializer. Now instead of the factory methods defaulting to throwing an error, they default to returning the base serializer and deserializer.

### CODE REVIEW

File: client/js/_author/components/assessments/question_types/_question.jsx [hunk 1]
```diff
@@ -99,6 +99,8 @@ export default class Question extends React.Component {
     const { name, type, id, question, bankId } = item;
     const className = this.getClassName();
 
+    const questionText = question ? question.text : '';
```
Comment (BenHSmith13) [inline client/js/_author/components/assessments/question_types/_question.jsx]: does file upload not have a question object?

File: client/js/middleware/serialization/qbank/serializers/base.js [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
+}
+
+export function baseSerializeItem(originalItem, newAttributes) {
+  const newItem = {
     id: originalItem.id,
     genusTypeId: genusTypes.item[originalItem.type],
-    name: newAttributes.name,
-    question: null,
+    name: newAttributes.name || originalItem.name,
+    question: baseSerializeQuestion(originalItem, newAttributes.question || {}),
```
Comment (BenHSmith13) [inline client/js/middleware/serialization/qbank/serializers/base.js]: should this question be scrubbed?

File: client/js/_author/components/assessments/question_types/_question.jsx (hunk 1 above)
Comment (nick-benoit14) [inline client/js/_author/components/assessments/question_types/_question.jsx]: This accounts for the question object being potentially null. Which the base deserializer defaults to.

File: client/js/middleware/serialization/qbank/serializers/base.js (hunk 2 above)
Comment (nick-benoit14) [inline client/js/middleware/serialization/qbank/serializers/base.js]: That happens on the next line. It should be on the same line though.

Comment (nick-benoit14) [inline client/js/middleware/serialization/qbank/serializers/base.js]: Or are you asking if it should not be scrubbed?

File: client/js/_author/components/assessments/question_types/_question.jsx (hunk 1 above)
Comment (BenHSmith13) [inline client/js/_author/components/assessments/question_types/_question.jsx]: Ah, thats fine. you can `_.get` or not, really either way is fine by me

File: client/js/middleware/serialization/qbank/serializers/base.js (hunk 2 above)
Comment (BenHSmith13) [inline client/js/middleware/serialization/qbank/serializers/base.js]: :] I must have missed the next line

**Comment by BenHSmith13** [comment]
lgtm

**Pull request merged**